### ISSUE TITLE
"You cannot be reached from outside." - It was working; now isn't. HW & SW firewalls unchanged.

### ISSUE THREAD

**Issue by Lew-Rockwell-Fan** [yacy/yacy_search_server#143]
I started getting this on /Status.html?noforward= yesterday:

"You cannot be reached from outside. A possible reason is that you are behind a firewall, NAT or Router."

My software and router firewall settings have been the same for a few days and were working fine 24 hours ago:

Software firewall:
Relevant output of "sudo ufw status verbose":

Status: active
Logging: on (low)
Default: deny (incoming), allow (outgoing), disabled (routed)
New profiles: skip

To                         Action      From
--                         ------      ----
8090                       ALLOW IN    Anywhere                  
8090 (v6)                  ALLOW IN    Anywhere (v6)

Router firewall setting is unchanged from when I first got this working. See attached screenshot.
![router_setup_re_yacy_2017-11-21_13-04-51](https://user-images.githubusercontent.com/16430650/33089848-ed92f030-cebf-11e7-9743-eabc492c6424.png)
Suggestions for figuring out what went wrong?

**Comment by luccioman** [comment]
Did you change the "Protection of all pages" setting in the /ConfigAccounts_p.html page? 
In the 1.92/9000 release this setting is "On", this makes your peer can not be reached from outside. But a [fix](https://github.com/yacy/yacy_search_server/commit/bd7411a53af4f1e54bf27498f95b4f0f7934f090) is available in the next developer releases.

**Comment by Lew-Rockwell-Fan** [comment]
> Did you change the "Protection of all pages" setting in the /ConfigAccounts_p.html page?
In the 1.92/9000 release this setting is "On", this makes your peer can not be reached from outside.

That setting was, and is, OFF. I don't recall changing it. 

From the Status.html?noforward= page my version is:
1.921/9342

I'm trying to keep one topic to a thread & address one issue at a time, but just in case any of these suggest why I might be having THIS problem:
-I'm still having very high cpu usage. I thought my newer machine had solved that problem, but apparently not.
-My crawler seems to be spontaneously shutting down in 2 different ways. Sometimes it acknowledges the shutdown with a message on Crawler_p.html like "I ran short of memory and paused." (not the exact wording :-)) and sometimes (like now) everything, including resource usage, indicates it is still running, but the Crawler PPM is 0 and no new page titles are appearing at the bottom of the page. Either way it starts working again if I restart it.

I repeat, I'm not trying to address those issues here and now. One thing at a time. I mention them just in case they could reflect on why my crawl results aren't open to the group.

Thanks.

**Comment by luccioman** [comment]
For sure, if your peer is running of out of resources, this may cause other peers mark it as unreachable because they get a HTTP error or a timeout when performing peer-to-peer presence protocol operations.

The OutOfMemory errors you get suggest that your YaCy JVM has not enough RAM to run fine. You can configure this in the "RAM/Disk Usage & Updates > Performance" page (/Performance_p.html), with the "Memory reserved for JVM" field. This page also lets you configure disk usage limitations.
Another setting can also be useful if you can not dedicate much memory to YaCy : the "Limit size of indexed remote results" field in the /ConfigPortal_p.html page can prevent adding too large documents from other peers to your own index.

One more note about SSL : even if not necessary for other peers to reach you, that's right that you have to open the 8443 port on your router if you want to let others use encrypted communication (with https) when requesting your peer (indeed, until [very recently](https://github.com/yacy/yacy_search_server/commit/f01aac31fdfd847c1a741a563f126efd5afd6149), this was mainly used for users who which to connect more securely to their YaCy installed on a remote server).

**Comment by Lew-Rockwell-Fan** [comment]
Thanks. I reset both the HW & SW firewalls maybe 12 hours ago to allow for https also, and bumped one of the RAM settings at the same time. I did crawl again successfully after that, so the RAM setting must have made a difference. Again halted for inadequate RAM when I looked a few minutes ago, so I bumped the same setting again & restarted yacy. I'd be more specific but  I'm waiting on the whale on Steering.html that says "Just a moment, please!
Application will terminate after working off all scheduled tasks.
Then YaCy will restart.
If you can't reach YaCy's interface after 5 minutes restart failed."
I believe it gets stuck on that  page even when it has restarted. I base  that belief on having waited much longer than 5 minutes in the past without change, only to go to one of the other yacy local pages I've bookmarked & find that the restart apparently WAS successful. I'll check back on it in a while after I do some essential meat space things.

**Comment by luccioman** [comment]
> 'Status.html?noforward=' page still says 'You cannot be reached from outside.'

Would you like to tell your peer name (here or in a private message on YaCy forum if you prefer) so one can do some checks on your public peer URL?

> it sounds like I'm setting a MINIMUM amount yacy will use

Not exactly. This is rather a maximum amount of memory used by the JVM running YaCy (see the related `-Xmxsize` section in the java executable [documentation](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/java.html) ). More especially when your own index is growing, YaCy is likely to use most part of this amount of memory because its internal data structures are designed to put as much as possible data in memory cache. That to prevent too much disk IO. In the end, that's right that this memory is made unavailable to other programs.

**Comment by Lew-Rockwell-Fan** [comment]
OK, you seem to imply there is some security issue in that. I don't see a facility for PM here. I finally managed to register at the forum. I don't give emails out to sites with bad security  certs & that still hasn't been corrected, so I used a trash email. Finally had to turn  off noscript.  What is  your nic there?

**Comment by luccioman** [comment]
@Lew-Rockwell-Fan I don't think it is a security issue to tell here the name of your peer. I just proposed you to use private forum messages for any reason relevant to you, for example if you would prefer to maintain some anonymity. Do as you wish.
By the way my user name on YaCy forum is "luc".

**Comment by Lew-Rockwell-Fan** [comment]
Thank you. Sent it there.

**Comment by luccioman** [comment]
Ok, so I tried yesterday and today requesting your peer with its reported IPv4 address, either on http or https port... but for now what I can only say it is that it always ends up in a timeout. Obviously that's why your peer is reported as "junior"/unreachable from outside.

Did you try to temporarily disable either your software firewall or your router one or both? This could at least help you identify if incoming connections are blocked or not by firewall rules.

**Comment by Lew-Rockwell-Fan** [comment]
> Did you try to temporarily disable either your software firewall or your router one or both? This could at least help you identify

I'll try turning off the SW firewall, ufw, again, for an hour or so today. Bypassing the hardware router will take more doing. I'll have to wait until I'm the only one in the building, otherwise I'd interrupt their connection. I'll try it ASAP.

Thanks.

**Comment by Lew-Rockwell-Fan** [comment]
Turning off ufw (software firewall) didn't make any difference. For now, I'm going to put the whole thing on hold while I set up some  other things and migrate  my main system from an ulltra-light Ubuntu 16.04 LTS to the Debian analog (stretch built from the net install iso).  The resource usage keeps growing & has become so problematic that it interferes with everything else  I do. So, when I  get  set up in Stretch, I'll seek advice on the forum about settings to minimize cpu usage & make it possible to run without dedicating half my RAM to it. Also, I'll have another system to compare. That alone could give some insight as to why my crawl results aren't available to peers. Then I'll come back to this. Probably in a couple of weeks.

Thanks for your help.

**Comment by luccioman** [comment]
Ok, one more tip that may help you : with Debian (and Ubuntu I guess), when you install YaCy using the Debian package, you can use systemctl to limit YaCy resources usage, notably regarding CPU. 

I mentioned it some time ago on the [forum](http://forum.yacy-websuche.de/viewtopic.php?f=18&t=5816&p=32992&hilit=+systemctl#p32992). At that time if was not yet effective with the SystemD version embedded in Debian Jessie, but I just checked with Debian Stretch and it looks like now SystemD perfectly does its job to limit the YaCy service CPU usage.

**Comment by zukunft** [comment]
I am having the same issue with Kubuntu 19.10. Yesterday it was working fine for a few hours, but today after a restart the connection does not work any more. Maybe a "retest now" button beside the port settings could be helpful.

**Comment by zukunft** [comment]
Update: if I open all ports for a short period, it starts working again. Is it possble, that UPnP uses an additional port?
